### DISCUSSION TITLE
🗄️ 一鍵查看你在GitHub上所有儲存庫使用了多少空間！

### DISCUSSION THREAD

**Discussion by xxrjun** [1101-GS4719-C/Python-Community-GS4719-C#1199] [優秀專案]
> 前天睡不著覺時突然好奇自己在GitHub上所有的Repos佔了多少空間 🗄️
>
> 網站網址 https://total-repos-size.netlify.app (部署於 Netlify)

## [xxrjun / total-repos-size](https://github.com/xxrjun/total-repos-size)

GitHub -> Settings -> Repositories 可以看每一個Repo的大小，但如果儲存庫很多個就得拿計算機一個一個加起來算。而且每次變動想查詢又要重算，所以做出一個計算的功能應該是比較合理的！

<img width="400" alt="截圖 2023-04-01 下午5 39 42" src="https://user-images.githubusercontent.com/40348319/229278506-a8409db8-8964-43e2-9bf8-f7b8b0770af1.png">

上網好像也沒找到一鍵顯示的功能，想說自己做可能也比找別人做好的還快，所以就串了一下 [GitHub REST API](https://docs.github.com/en/rest) 完成一個簡單的小網站，在這邊與大家分享 :mechanical_leg:

使用方式也很簡單，只要輸入GitHub ID就能夠一鍵查看自己在GitHub上所有儲存庫使用了多少空間！

如下圖所示，我在GitHub上有 **30** 個public repos、共使用 **331MB** 左右，而 @benctw  老師有 **29** 個public repos、共使用 **1.86GB** 左右

<img width="600" alt="截圖 2023-04-01 下午5 52 07" src="https://user-images.githubusercontent.com/40348319/229279269-43851720-e41b-4f92-80f5-883923e1b2b4.png">

這個網站有單人模式也有雙人模式的頁面，可以跟朋友比比看 :raised_hands:

另外需要注意這個網站只會計算公開儲存庫的大小，且並非下載(clone)後的大小。
因為官方文件沒有寫，所以查了一下stackoverflow，上面說這是壓縮過存在GitHub的大小，會壓縮差不多60%的大小。

## Built with

![Svelte](https://img.shields.io/badge/Svelte-4A4A55?style=for-the-badge&logo=svelte&logoColor=FF3E00) ![SvelteKit](https://img.shields.io/badge/SvelteKit-FF3E00?style=for-the-badge&logo=Svelte&logoColor=white) ![GitHub](https://img.shields.io/badge/GitHub-100000?style=for-the-badge&logo=github&logoColor=white) ![Netlify](https://img.shields.io/badge/Netlify-00C7B7?style=for-the-badge&logo=netlify&logoColor=white)

|             類別              |                     名稱                     |
| --------------------------- | ------------------------------------------ |
| 前端框架 (Frontend Framework) | [Svelte](https://github.com/sveltejs/svelte) |
| API 串接 (API Integration) | [GitHub REST API](https://docs.github.com/en/rest) |
| 部署 (Deploy)| [Netlify](https://www.netlify.com/) |
| 其他工具 (Others)| [ChatGPT](https://openai.com/blog/chatgpt)、[GitHub Copilot](https://github.com/features/copilot) |

## 心得

過去寫前端都是用React框架，最近想說嘗試看看近幾年備受熱愛的 svelte，不過這個小project都是SPA所以就只是小體驗一下 :monocle_face:。一開始先看Svelte官方文件，建立起一個Svelte應用。列出**使用工具、系統功能、需要串接的API**給ChatGPT生出程式碼後，稍微調一下架構，差不多30分鐘就完成了。真心推薦大家開發時多多利用ChatGPT以及GitHub Copilot等工具！

後續再增添一些Dark & Light Mode、單雙人頁面比對等功能，還有排版，差不多要花到五個、六小時，整體而言我覺得還是蠻快的。這次開發的過程中，我發現能夠下出精準的prompt對我而言是一門很深的學問。常常一開始罵ChatGPT產出的東西根本是 :hankey: ，結果把prompt修的更精準後，產出的結果不但符合需求，而且遠比自己打還快。

另外我覺得Html, JavaScript或是一些框架的語法都難不太倒ChatGPT以及GitHub Copilot，但CSS的部分真的很難，可能因為樣式很主觀加上我自己不太知道如何下模型能夠懂的指令。像是我可能想要一個很有科技、有質感的Card Component，科技、有質感這類型的形容詞都有點太抽象了。我想能夠將抽象化為具體也是一種難以被科技起代的能力吧! :thinking:

**Comment by Glacier-Luo** [comment]
好工具，很不错。
